### ISSUE TITLE
Add possibility to apply suggestion(s)

### ISSUE THREAD

**Issue by jakubbortlik** [harrisoncramer/gitlab.nvim#393]
## Feature Description

One good thing from gitlab online that is missing in `gitlab.nvim` is applying suggestions. Applying a suggestion automatically closes the thread, which streamlines the review process (reviewer does not have to check whether MR author applied the suggestion as recommended.

I'm not sure if [go-gitlab](https://github.com/xanzy/go-gitlab) actually supports this. If not, it could possibly be implemented locally - after all, it's just applying the suggested change, staging, creating a commit and pushing to remote.

There'd be some catches:
1. It would be nice to see a preview of the suggestion (this would ideally be a separate issue request)
2. It would be nice to be able to add a suggestion to a batch that would be applied in one commit later or directly apply a single suggestion.
3. Gitlab makes it possible to add several suggestions in one comment and to select which one the user wants to apply: 
![image](https://github.com/user-attachments/assets/4f9769bf-1eb0-4893-b7ee-a1aa369fbc52)

**Comment by harrisoncramer** [comment]
We may do this eventually but it's pretty involved so not high on the priority list. Managing that state and applying suggested edits is a fair bit of work, and I'm not inclined to focus on this feature at the moment. 

That said, if you wanted to do some research into a proposal and drop it here I'd be happy to give feedback.

**Comment by jakubbortlik** [comment]
You are right, that having the plugin apply suggestions would be complicated. I've realized, however, that this is not really necessary. I've started working on a simple way to preview suggestions. The main goal is basically just to automate some of the actions the users have to do themselves:
1. get the file name and suggestion range
2. get the original version on which the suggestion was made (if this version is identical with HEAD, then just use HEAD (in the spirit of Diffview's `imply_local`)
3. create a scratch buffer with the suggestion applied
4. open a new tab with a normal Nvim diff of these two buffers.

The users can then easily take it from there and stage the selected changes with the tools they already use (e.g., `Gitsigns` in my case).

My original approach was to create a separate buffer for each suggestion in a note (I guess there are rarely more than one suggestion per comment, but this is possible in Gitlab, and I don't think it add too much complexity to the plugin to support it) and the diff it against the local HEAD. This is not ideal as the user then has to manually copy the changes from the suggestion buffer to the local HEAD file if they want to have proper LSP support on it, run tests with the change applied etc.

I'm going to try a different approach now, I'd like to create a scratch buffer with the original version and then make it possible to easily apply the suggestion(s) on the local HEAD file and easily discard those changes.

In the next step I'd like to make it possible for the user to use a similar approach when *creating* or *modifying* a suggestion - I'd like to make it equally easy to prepare a suggestion with the support of LSP/Treesitter and to be able to run tests with the suggestion applied, etc.
